src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs

```diff
@@ -1325,20 +1359,26 @@ bool WaitingForRememberEntitiesStore(object message)
                     case RememberEntityStoreCrashed msg:
                         ReceiveRememberEntityStoreCrashed(msg);
                         return true;
-                    case var msg when _extractEntityId(msg).HasValue:
-                        DeliverMessage(msg, Sender);
-                        return true;
-                    case var msg:
-                        // shouldn't be any other message types, but just in case
-                        Log.Warning("{0}: Stashing unexpected message [{1}] while waiting for remember entities update of starts [{2}], stops [{3}]",
-                            _typeName,
-                            msg.GetType().Name,
-                            string.Join(", ", update.Started),
-                            string.Join(", ", update.Stopped));
-                        Stash.Stash();
+                    default:
+                        var entityId = _extractor.EntityId(message);
```

Comment:
We only do this check here for catching bugs - otherwise these messages will get stashed and processed again later.